subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -138,10 +140,17 @@ static char *sprint_token(const u8_t *token, u8_t tkl)
 	static char buf[32];
 	int pos = 0;
 
-	for (i = 0; i < tkl; i++) {
-		pos += snprintf(&buf[pos], 31 - pos, "%x", token[i]);
+	if (token && tkl != LWM2M_MSG_TOKEN_LEN_SKIP) {
+		for (i = 0; i < tkl; i++) {
```

Comment:
you could move 'i' variable declaration inside that if statement.

(and aerate you code, line breaks are cheap it's nice to add an empty line after a } unless it's another } etc...)